src/main/java/com/Elessar/app/util/kafka/ProducerCreator.java

```diff
@@ -14,6 +14,9 @@
         Properties props = new Properties();
         props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
         props.put(ProducerConfig.CLIENT_ID_CONFIG, "localhost:9092");
+        props.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, true);  // exactly once delivery => require in-flight requests < 5, retries > 0 and ack = all
```

Comment:
Can you provide a link to where those requirements come from?

Reply:
>enable.idempotence
When set to 'true', the producer will ensure that exactly one copy of each message is written in the stream. If 'false', producer retries due to broker failures, etc., may write duplicates of the retried message in the stream. **Note that enabling idempotence requires max.in.flight.requests.per.connection to be less than or equal to 5, retries to be greater than 0 and acks must be 'all'.** If these values are not explicitly set by the user, suitable values will be chosen. If incompatible values are set, a ConfigException will be thrown.
https://docs.confluent.io/current/installation/configuration/producer-configs.html